src/pal/src/include/pal/stackstring.hpp

```diff
@@ -44,34 +52,42 @@ class StackString
             return;
         }
 
+        if(preserve)
```

Comment:
Actually i am getting out of this preserve business and am always preserving now by virtue of using realloc